### PR TITLE
fix: Properly set team on login

### PR DESCRIPTION
**Pull request by candiduslynx**
Closes https://github.com/cloudquery/cloudquery/issues/16298

The following scenarios have to be taken care of:
1. The user explicitly selects team via `--team` flag
2. The user has previously selected team
   1. The team is available after login – use it
   2. The team is unavailable – use the opt 3
3. Check that 1 & only 1 team is available for the user – & use it

### CODE REVIEW

File: cli/cmd/login.go [hunk 1]
```diff
@@ -177,60 +177,78 @@ func runLogin(ctx context.Context, cmd *cobra.Command) (err error) {
...
+	// list all available teams
+	teams, err := cl.ListAllTeams(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to list teams: %w", err)
 	}
+
 	if cmd.Flags().Changed("team") {
-		selectedTeam := cmd.Flag("team").Value.String()
-		err = cl.ValidateTeam(ctx, selectedTeam)
+		// don't care about the current value
```
Comment (erezrokah) [inline cli/cmd/login.go:214]:
```suggestion
		// don't care about the current cached config value as the user explicitly passes the `team` flag
```

File: cli/cmd/login.go [hunk 2]
```diff
@@ -177,60 +177,78 @@ func runLogin(ctx context.Context, cmd *cobra.Command) (err error) {
...
+		cmd.Printf("Your current team is set to %s.\n", currentTeam)
+		return nil
+	}
 
-	return nil
+	// 0 or > 1 teams available
+	cmd.Printf("Your current team is not set.\n\n")
+	cmd.Printf("Teams available to you: " + strings.Join(teams, ", ") + "\n\n")
+	cmd.Printf("To set your current team, run `cloudquery switch <team>`\n\n")
+	return fmt.Errorf("team not set. Available teams: [" + strings.Join(teams, ", ") + "]")
```
Comment (erezrokah) [inline cli/cmd/login.go:253]: Did we fail before if teams > 1? Not sure we should fail the login command. The user will get another error message if they use a command that requires a team to be set

File: cli/cmd/login.go [hunk 3]
```diff
@@ -177,60 +177,78 @@ func runLogin(ctx context.Context, cmd *cobra.Command) (err error) {
...
+			return fmt.Errorf("failed to set team: %w", err)
+		}
+		cmd.Printf("Your current team is set to %s.\n", currentTeam)
+		return nil
+	}
 
-	return nil
+	// 0 or > 1 teams available
+	cmd.Printf("Your current team is not set.\n\n")
+	cmd.Printf("Teams available to you: " + strings.Join(teams, ", ") + "\n\n")
```
Comment (erezrokah) [inline cli/cmd/login.go:251]: If the user doesn't have any teams we'll print an empty array. Maybe we should direct the user to create a team on the cloud Website instead?

**Review by erezrokah — CHANGES_REQUESTED** [review]
Great change, added a few comments

File: cli/cmd/login.go (hunk 3 above)
Comment (candiduslynx) [reply cli/cmd/login.go:251]: I think that would be a separate change unrelated to the fixed bug. I can do that as a follow-up

File: cli/cmd/login.go (hunk 2 above)
Comment (candiduslynx) [reply cli/cmd/login.go]: I removed error in ea441bb43e09d71ade958d3711e3ea768e9b0d30

**Review by erezrokah — APPROVED** [review]

**Pull request merged**